### PR TITLE
Compose Era!!!

### LINKED ISSUE #342: 返回时左侧菜单没有更新
### 复现步骤 / Step to reproduce

1. 进入应用，这是默认应该在主页。
2. 打开左侧菜单，进入热门。
3. 滑动返回或者点击返回键。
4. 打开左侧菜单。

### 预期行为 / Expected behaviour

高亮显示主页按键。

### 实际行为 / Actual Behaviour

高亮仍然在刚才的热门。

### 应用日志 / App logs

N/A

### 备注 / Addition details

_No response_

### 浏览站点 / Browsing site

ExHentai

### EhViewer 版本号 / EhViewer version code

1.8.12.0-SNAPSHOT

### EhViewer CI 版本 / EhViewer CI Version

0aba590

### Android 系统版本 / Android version

N/A

### 设备型号 / Device model

N/A

### SoC 型号 / Soc model

N/A

### 自查步骤 / Verify steps

- [ ] 如果您有足够的时间和能力，并愿意为此提交 PR ，请勾上此复选框 / Pull request is welcome. Check this if you want to start a pull request
- [X] 您已搜索过 [Issue Tracker](https://github.com/FooIbar/EhViewer/issues)，没有找到类似内容 / I have searched on [Issue Tracker](https://github.com/FooIbar/EhViewer/issues), No duplicate or related open issue has been found
- [X] 您确保这个 Issue 只提及一个问题。如果您有多个问题报告，烦请发起多个 Issue / Ensure there is only one bug report in this issue. Please make mutiply issue for mutiply bugs
- [X] 您确保已使用最新 [CI 版本](https://github.com/FooIbar/EhViewer/actions)测试，并且该问题在最新 CI 版本中并未解决 / This bug have not solved in latest [CI version](https://github.com/FooIbar/EhViewer/actions)

### PR DESCRIPTION
**Pull request by VDostoyevskiy**
Resolve #342

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryDetailFragment.kt [hunk 1]
```diff
@@ -942,10 +943,19 @@ fun GalleryDetailScreen(args: GalleryDetailScreenArgs, navigator: DestinationsNa
             lub.keyword = uploader
             navigator.navigate(GalleryListScreenDestination(lub))
         }
+
+        var showBottomSheet by remember { mutableStateOf(false) }
+
+        if (showBottomSheet && galleryDetail != null) {
+            ModalBottomSheet(onDismissRequest = { showBottomSheet = false }) {
+                GalleryInfoBottomSheet(galleryDetail, navigator)
+            }
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryDetailFragment.kt:956]: https://issuetracker.google.com/issues/278679016
https://issuetracker.google.com/issues/281967264

Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/GalleryDetailFragment.kt:956]: > https://issuetracker.google.com/issues/278679016 https://issuetracker.google.com/issues/281967264

Got it, will grab BottomSheet in next CL

**Comment by VDostoyevskiy** [comment]
@FooIbar 
Let's squash merge this now
Too much conflicts

**Review by FooIbar — APPROVED** [review]

**Comment by VDostoyevskiy** [comment]
Later, we make SnackBarState CompositionLocal, make NavDrawer/SideSheet CompositionLocal

File: app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt [hunk 2]
```diff
@@ -137,20 +140,16 @@ import splitties.systemservices.clipboardManager
...
-    Triple(R.id.nav_history, R.string.history, Icons.Default.History),
-    Triple(R.id.nav_downloads, R.string.downloads, Icons.Default.Download),
-    Triple(R.id.nav_settings, R.string.settings, Icons.Default.Settings),
+    Triple(HomePageScreenDestination, R.string.homepage, Icons.Default.Home),
+    Triple(SubscriptionScreenDestination, R.string.subscription, EhIcons.Default.Subscriptions),
+    Triple(WhatshotScreenDestination, R.string.whats_hot, Icons.Default.Whatshot),
+    Triple(ToplistScreenDestination, R.string.toplist, Icons.Default.FormatListNumbered),
+    Triple(FavouritesScreenDestination, R.string.favourite, Icons.Default.Favorite),
+    Triple(HistoryScreenDestination, R.string.history, Icons.Default.History),
+    Triple(DownloadScreenDestination, R.string.downloads, Icons.Default.Download),
```
[the remark this answers is not in the document]
Comment (VDostoyevskiy) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/MainActivity.kt:149]: LMAO

**Comment by FooIbar** [comment]
https://issuetracker.google.com/issues/304435200

**Pull request merged**